network/peer/tls_config.go

```diff
@@ -26,5 +42,38 @@ func TLSConfig(cert tls.Certificate, keyLogWriter io.Writer) *tls.Config {
 		InsecureSkipVerify: true, //#nosec G402
 		MinVersion:         tls.VersionTLS13,
 		KeyLogWriter:       keyLogWriter,
+		VerifyConnection:   ValidateCertificate,
+	}
+}
+
+// ValidateCertificate validates TLS certificates according their public keys on the leaf certificate in the certification chain.
+func ValidateCertificate(cs tls.ConnectionState) error {
+	if len(cs.PeerCertificates) == 0 {
+		return ErrNoCertsSent
+	}
+
+	if cs.PeerCertificates[0] == nil {
+		return ErrEmptyCert
+	}
+
+	pk := cs.PeerCertificates[0].PublicKey
+
+	switch key := pk.(type) {
+	case ed25519.PublicKey:
+		return ErrForbidden25519Key
+	case *ecdsa.PublicKey:
+		if key == nil {
+			return ErrEmptyPublicKey
+		}
+		if key.Curve != elliptic.P256() {
+			return ErrCurveMismatch
+		}
+		return nil
+	case *rsa.PublicKey:
+		return staking.ValidateRSAPublicKeyIsWellFormed(key)
+	case nil:
+		return ErrEmptyPublicKey
+	default:
+		return nil
 	}
```

Comment:
Feels like we can reduce the number of cases in this flow by only specifying the types that we want to allow.
```suggestion
	switch key := pk.(type) {
	case *ecdsa.PublicKey:
		if key == nil {
			return ErrEmptyPublicKey
		}
		if key.Curve != elliptic.P256() {
			return ErrCurveMismatch
		}
		return nil
	case *rsa.PublicKey:
		return staking.ValidateRSAPublicKeyIsWellFormed(key)
	default:
		return ErrUnsupportedKeyType
	}
```

Reply:
[Copied](https://github.com/ava-labs/avalanchego/pull/3501/commits/e45a72ae90a2c993915320789a299e78557fccac#diff-566b013fa8a33a9c2561ca4b8951d78da4fa0bb3f63fc3b48499d425f9c80217L60-L77) your code snippet.